### ISSUE TITLE
Dragging items through one another is ugly

### ISSUE THREAD

**Issue by celiasmith** [nengo/nengo_viz#244]
This is especially the case when you're dragging something through a nested network.  Some populations go on top of the item being dragged, some below, etc.  

I guess the right thing to do is have the dragged item come to the top while being dragged.

What happens when you let go is not obvious.  Should it stay on top of where it was released? (Will this cause possible stacking problems with z_indexes?)  Probably don't want it to disappear behind where it is released... (or return to it's previous level, which could cause that).

**Comment by tcstewar** [comment]
> I guess the right thing to do is have the dragged item come to the top while being dragged.

The problem is when you're dragging an expanded Network.  If you make the Network come to the top (and all of the things inside it), any of the Connections to something outside the Network will now be behind the network.  And if you make those Connections come to the front as well, then they'll be on top of the things they are connected to.  So I don't know what this should look like.  :(

> What happens when you let go is not obvious. Should it stay on top of where it was released? (Will this cause possible stacking problems with z_indexes?) Probably don't want it to disappear behind where it is released... (or return to it's previous level, which could cause that).

That's not a problem, since SVG doesn't do z-indexes.  It's just drawing things in order, and you move things to the top by putting them at the end of the draw order.  So just leaving things how they lokoed while you were dragging should be fine.

**Comment by celiasmith** [comment]
Yeah this is a tough one.  Here's a possible, hopefully low-overhead option: Turn the links coming into the network to grey or dashed while it's being dragged.  This would pop them on top of their externally connected objects, but the visual change would make clear that this was temporary, and to be sorted out on click release.  On release, you could hide any connections where the middle of the external object was under the dragged item.

**Comment by tcstewar** [comment]
This is definitely better than it was before (thanks to #245), but there's definitely still situations where it looks weird.  However, there isn't a clear thing to do to improve it.  If someone wants to take a stab at this, it'd be useful to identify a few specific situations where things look wrong and exactly what it ought to look like.  Then we might be able to sort out how to adjust the algorithm.
